test/camera/test_camera.cpp

```diff
@@ -0,0 +1,444 @@
+/*******************************************************************************
+ * Copyright (c) 2018-2024 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *******************************************************************************/
+
+#include <fstream>
+
+#include <Camera/Camera.h>
+#include <ECS/Registry.h>
+#include <Locator.h>
+#include <gtest/gtest.h>
+#include <json_helpers.h>
+
+#include "scenarios/DoubleClickFlyTo.h"
+#include "scenarios/DragUpDown.h"
+#include "scenarios/MiddleDragRightUp.h"
+#include "scenarios/MoveBackwardForward.h"
+#include "scenarios/MoveRightLeft.h"
+#include "scenarios/PanRightLeft.h"
+#include "scenarios/TiltDownZoomOut.h"
+#include "scenarios/TiltUpDown.h"
+#include "scenarios/TiltUpPanLeft.h"
+#include "scenarios/TwoButtonZoomOutIn.h"
+#include "scenarios/ZoomOutIn.h"
+
+// Enable this define because we use a custom locator
+#define LOCATOR_IMPLEMENTATIONS // TODO: If needed put this in cmake for all tests
+#include <Audio/AudioManagerNoOp.h>
+#include <Camera/DefaultWorldCameraModel.h>
+#include <Common/RandomNumberManagerTesting.h>
+
+#include "scenarios/Mock.h"
+
+using nlohmann::json;
+using openblack::ecs::Registry;
+using namespace openblack;
+
+struct TestValues
+{
+	std::string name;
+	MockDynamicsSystem* dynamicsSystem;
+	MockAction* actionInterface;
+};
+
+class TestDefaultCameraModel: public testing::TestWithParam<TestValues>
+{
+protected:
+	void SetUp() override
+	{
+		const auto testName = GetParam().name;
+		const auto testResultsPath = std::filesystem::path(k_ScenarioPath) / (testName + std::string(".json"));
+		if (!std::filesystem::exists(testResultsPath))
+		{
+			GTEST_SKIP() << "Unable to find the scenario test json, is it in the path?";
```

Comment:
Perhaps we should either have the json exported with the test or include them in the binary